app/controllers/publishers_controller.rb

```diff
@@ -266,8 +266,6 @@ def get_site_banner_data
   private
 
   def authenticate_via_token
-    sign_out(current_publisher) if current_publisher
```

Comment:
Could this raise an exception if we continue down with the method with an already signed in user? 

Perhaps we should just return if they are already authenticated? (Not sure about that one)

Reply:
I tested by: 
(0) Open mailcatcher in non-private window
(1) Signing in a non-private window for a@a.com
(2) Fetching a login email in a private window for b@b.com
(3) In non-private window, clicking on the link in mailcatcher for b@b.com
(4) See that session has switched to b@b.com